demos/form-control/Slider.php

```diff
@@ -0,0 +1,215 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Ui\Form\Control;
+
+use Atk4\Ui\View;
+use Atk4\Ui\Js\JsExpression;
+use Atk4\Ui\Js\JsFunction;
+use Atk4\Ui\Js\JsReload;
+
+class Slider extends Input
+{
+    public string $inputType = 'hidden';
+
+    /** The lowest value the slider can be. */
+    public int $min = 0;
+
+    /** The max value the slider can be. */
+    public int $max = 10;
+
+    /** @var float|null The slider step. Set to 0 to disable step. */
+    public $step = 0;
```

Comment:
Just a question for now - how feasible is histogram below the slider and non-liner steps or even non-numeric values (XS-S-M-L...)?

Reply:
Yes, it's possible!
According to Fomantic [https://fomantic-ui.com/modules/slider.html#custom-interpreted-labels](https://fomantic-ui.com/modules/slider.html#custom-interpreted-labels) it can be done with:

```
var labels = ["α", "β", "γ", "δ", "ε", "ζ", "η", "θ", "ι", "κ", "λ", "μ", "ν", "ξ", "ο", "π", "ρ", "σ/ς", "τ", "υ", "φ", "χ", "ψ", "ω"];
$('#interpretedlabel')
  .slider({
    interpretLabel: function(value) {
      return labels[value];
    }
  })
;
```
As a matter of fact, all the options I've implemented, except this one is directly from Fomantic. I just broke out ticked, bottom align and labeled from the class and made those options. If some of them are funky, yes, I know, they can be removed if we deem the don't add anything. I just added it for completness for now.

Reply:
@mvorisek, I think I need your help on this one. It seems that the function (interpretLabel) need to be existing before  semantic.js (Fomantic-UI) is loaded. I've hacked a version of semantic.js to try this and it indeed added the custom labels then. This is not acceptable though as I know very well, it was just done for test.
I do need help though to add the function first on the page before loading Fomantic-UI. Maybe there is another way but I've already spent to much time on this without any workable solution.

Reply:
> It seems that the function (interpretLabel) need to be existing before semantic.js (Fomantic-UI) is loaded.

That seems weird to me. All user JS is evaluated "after page is fully loaded", so the elements are guaranteed to exists. And you can define a JS function before you setup the FUI slider by JS.

I do not understand the issue you are facing here.. If you need to define global JS code - to deduplicate long JS code - you can put the code in `/js/*.js`. If you need a small JS code per control, you should always use anonymous JS function.

Reply:
@mvorisek I fixed it... Using JsFunction and JsExpression was the way to go but I had trouble with the array but using slice instead of addressing the item directly solved it.